assets/app/scripts/controllers/build.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:PodController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('BuildController', function ($scope, $routeParams, DataService, project, BuildsService, $filter) {
+    $scope.build = null;
+    $scope.buildConfigName = $routeParams.buildconfig;
+    $scope.builds = {};
+    $scope.alerts = {};
+    $scope.renderOptions = $scope.renderOptions || {};    
+    $scope.renderOptions.hideFilterWidget = true;    
+    $scope.breadcrumbs = [
+      {
+        title: "Builds",
+        link: "project/" + $routeParams.project + "/browse/builds"
+      },
+      {
+        title: $routeParams.buildconfig,
+        link: "project/" + $routeParams.project + "/browse/builds/" + $routeParams.buildconfig
+      },      
+      {
+        title: $routeParams.build
+      }
+    ];
+
+    var watches = [];
+    var objectWatches = [];
+
+    project.get($routeParams.project).then(function(resp) {
+      angular.extend($scope, {
+        project: resp[0],
+        projectPromise: resp[1].projectPromise
+      });
+      DataService.get("builds", $routeParams.build, $scope).then(
+        // success
+        function(build) {
+          $scope.build = build;
+          var buildNumber = $filter("annotation")(build, "buildNumber");
+          if (buildNumber) {
+            $scope.breadcrumbs[2].title = "#" + buildNumber;
+          }
+
+          // If we found the item successfully, watch for changes on it
+          objectWatches.push(DataService.watchObject("builds", $routeParams.build, $scope, function(build, action) {
```

Comment:
pretty sure it does allow watching a single object, but starting to open
websockets to just watch a single item complicates the logic in data.js,
what if I'm watching a single item, but then I want to start watching all
items of that type, do I close the websocket for the single item and
re-open to watch all, what if i then stop watching the full list but im
still watching the single item, again do i close the one websocket and
re-open the other?  We already have logic around whether a list operation
is already in progress and a watch is requested that we use the existing
list op, if are doing a watch and want to do a get to start and a list is
in progress what do we do.. etc etc etc  @liggitt and I decided to punt on
it for this sprint and define what the JS API would be for the method and
just reuse the existing watch behavior under the covers.  Its on my todo
list for next sprint to see if we can do better.

This would be a lot easier if there was an API where we could open a single
websocket, and then send requests for things to start watching.
 @smarterclayton and I discussed this like a year ago

On Thu, Sep 24, 2015 at 1:08 PM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/scripts/controllers/build.js
> <https://github.com/openshift/origin/pull/4726#discussion_r40344539>:
>
> > +    project.get($routeParams.project).then(function(resp) {
> > +      angular.extend($scope, {
> > +        project: resp[0],
> > +        projectPromise: resp[1].projectPromise
> > +      });
> > +      DataService.get("builds", $routeParams.build, $scope).then(
> > +        // success
> > +        function(build) {
> > +          $scope.build = build;
> > +          var buildNumber = $filter("annotation")(build, "buildNumber");
> > +          if (buildNumber) {
> > +            $scope.breadcrumbs[2].title = "#" + buildNumber;
> > +          }
> > +
> > +          // If we found the item successfully, watch for changes on it
> > +          objectWatches.push(DataService.watchObject("builds", $routeParams.build, $scope, function(build, action) {
>
> Also right now watchObject doesn't just watch the object, it watches the
> whole resource type under the covers.
>
> I noticed that as well... Does k8s not allow you to watch individual
> resources?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4726/files#r40344539>.
>